AddInstallationScriptTest.cmake

```diff
@@ -0,0 +1,25 @@
+macro(add_installation_script_test name path)
+
+  # Copy the source to the binary tree 
+  configure_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/${path}/${name}
+    ${CMAKE_CURRENT_BINARY_DIR}/${path}/${name}
+    COPYONLY
+  )
+  configure_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/${path}/${name}-usage
+    ${CMAKE_CURRENT_BINARY_DIR}/${path}/${name}-usage
+    COPYONLY
+  )
+  set(harness "${CMAKE_BINARY_DIR}/staging/test-${name}.sh")
+  install(
+      FILES "${harness}"
+      PERMISSIONS WORLD_EXECUTE WORLD_READ WORLD_WRITE OWNER_EXECUTE OWNER_READ OWNER_WRITE GROUP_EXECUTE GROUP_READ GROUP_WRITE
+      DESTINATION ${CMAKE_BINARY_DIR}/${path}
+  )
```

Comment:
This is actually the cause, AFAICT of the failures I'm seeing on Travis-CI. This line enforces the requirement of CTest *AFTER* `make install` which is highly unusual, since the canonical order is something like:

```
cmake|configure
make
make check|make test|ctest
make install
```

I have what I believe to be a good work around.